config/query_log.go

```diff
@@ -41,3 +44,25 @@ func (c *QueryLog) LogConfig(logger *logrus.Entry) {
 	logger.Infof("flushInterval: %s", c.FlushInterval)
 	logger.Infof("fields: %s", c.Fields)
 }
+
+func (c *QueryLog) censoredTarget() string {
+	// Make sure there's a scheme, otherwise the user is parsed as the scheme
+	targetStr := c.Target
+	if !strings.Contains(targetStr, "://") {
+		targetStr = c.Type.String() + "://" + targetStr
+	}
+
+	target, err := url.Parse(targetStr)
+	if err != nil {
+		return c.Target
+	}
+
+	if target.User == nil {
+		return c.Target
+	}
+
+	// Drop the password since special chars like * get URL escaped
+	target.User = url.User(target.User.Username())
```

Comment:
If the password is completly droped it might seem like it isen't set at all which could indicate a configuration error.

It might be better to just replace the password at string level
```suggestion
	if pass, hasPass :=target.User.Password(); hasPass {
		return strings.Replace(target.String(), pass, strings.Repeat("*", len(pass)), 1)
	}
```

Reply:
Good point!